nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -688,11 +700,15 @@ synchronized Scheduler prepareScheduler(final ProcessContext context, final Proc
     private ShardRecordProcessorFactory prepareRecordProcessorFactory(final ProcessContext context, final ProcessSessionFactory sessionFactory) {
         return () -> {
             if (isRecordReaderSet && isRecordWriterSet) {
+                final String value = context.getProperty(OUTPUT_STRATEGY).getValue();
```

Comment:
This should be changed to use `asAllowableValue(OutputStrategy.class)` so that the comparison against the `enum` can be used.